libraries/common-arm/src/sd_manager.rs

```diff
@@ -52,85 +56,126 @@ where
 {
     pub fn new(spi: SPI, cs: CS) -> Self {
         let time_sink: TimeSink = TimeSink::new(); // Need to give this a DateTime object for actual timing.
-        let mut sd_cont = sd::Controller::new(sd::SdMmcSpi::new(spi, cs), time_sink);
-        match sd_cont.device().init() {
-            Ok(_) => match sd_cont.device().card_size_bytes() {
+        let mut sd_controller = sd::Controller::new(sd::SdMmcSpi::new(spi, cs), time_sink);
+        match sd_controller.device().init() {
+            Ok(_) => match sd_controller.device().card_size_bytes() {
                 Ok(size) => info!("Card is {} bytes", size),
-                Err(_) => panic!("Cannot get card size"),
+                Err(_) => error!("Cannot get card size"),
             },
             Err(_) => {
-                panic!("Cannot get SD card.");
-            }
-        }
-
-        let mut volume = match sd_cont.get_volume(sd::VolumeIdx(0)) {
-            Ok(volume) => volume,
-            Err(_) => {
-                panic!("Cannot get volume 0");
+                herror!(Error, ErrorContext::SDCardNotConnected);
             }
         };
 
-        let root_directory = match sd_cont.open_root_dir(&volume) {
-            Ok(root_directory) => root_directory,
+        let mut volume = match sd_controller.get_volume(sd::VolumeIdx(0)) {
+            Ok(volume) => Some(volume),
             Err(_) => {
-                panic!("Cannot get root");
+                error!("Cannot get volume 0");
+                None
             }
         };
-        let file = sd_cont.open_file_in_dir(
-            &mut volume,
-            &root_directory,
-            "log.txt",
-            sd::Mode::ReadWriteCreateOrTruncate,
-        );
-        let file = match file {
-            Ok(file) => file,
+
+        let mut root_directory: Option<Directory> = None;
+        if volume.is_some() {
+            root_directory = match sd_controller.open_root_dir(volume.as_ref().unwrap()) {
+                Ok(root_directory) => Some(root_directory),
+                Err(_) => {
+                    error!("Cannot get root");
+                    None
+                }
+            };
+        }
+
+        let file: Result<sd::File, sd::Error<sd::SdMmcError>>;
+        if volume.is_some() && root_directory.is_some() {
+            file = sd_controller.open_file_in_dir(
+                volume.as_mut().unwrap(),
```

Comment:
@NoahSprenger Since there are checks for the optionals being populated the `unwrap`s should be guaranteed to not panic but is there a safer way to handle this? 
Same question for when unwrap is used down in the other methods